chain/arweave/src/adapter.rs

```diff
@@ -90,3 +114,94 @@ impl ArweaveBlockFilter {
         self.trigger_every_block = self.trigger_every_block || other.trigger_every_block;
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::sync::Arc;
+
+    use graph::{prelude::Link, semver::Version};
+    use sha256::digest_bytes;
+
+    use crate::data_source::{DataSource, Mapping, Source, TransactionHandler};
+
+    use super::{ArweaveTransactionFilter, HEX_SHA256_LEN};
+
+    const ARWEAVE_PUBKEY_EXAMPLE: &str = "x-62w7g2yKACOgP_d04bhG8IX-AWgPrxHl2JgZBDdNLfAsidiiAaoIZPeM8K5gGvl7-8QVk79YV4OC878Ey0gXi7Atj5BouRyXnFMjJcPVXVyBoYCBuG7rJDDmh4_Ilon6vVOuHVIZ47Vb0tcgsxgxdvVFC2mn9N_SBl23pbeICNJZYOH57kf36gicuV_IwYSdqlQ0HQ_psjmg8EFqO7xzvAMP5HKW3rqTrYZxbCew2FkM734ysWckT39TpDBPx3HrFOl6obUdQWkHNOeKyzcsKFDywNgVWZOb89CYU7JFYlwX20io39ZZv0UJUOEFNjtVHkT_s0_A2O9PltsrZLLlQXZUuYASdbAPD2g_qXfhmPBZ0SXPWCDY-UVwVN1ncwYmk1F_i35IA8kAKsajaltD2wWDQn9g5mgJAWWn2xhLqkbwGbdwQMRD0-0eeuy1uzCooJQCC_bPJksoqkYwB9SGOjkayf4r4oZ2QDY4FicCsswz4Od_gud30ZWyHjWgqGzSFYFzawDBS1Gr_nu_q5otFrv20ZGTxYqGsLHWq4VHs6KjsQvzgBjfyb0etqHQEPJJmbQmY3LSogR4bxdReUHhj2EK9xIB-RKzDvDdL7fT5K0V9MjbnC2uktA0VjLlvwJ64_RhbQhxdp_zR39r-zyCXT-brPEYW1-V7Ey9K3XUE";
+
+    #[test]
+    fn transaction_filter_some_empty_matches_all() {
+        let dss = vec![
+            new_datasource(None, 10),
+            new_datasource(Some("".into()), 10),
+            new_datasource(Some(ARWEAVE_PUBKEY_EXAMPLE.into()), 10),
+            new_datasource(
+                Some("2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824".into()),
```

Comment:
I expect developers to rather use their the short form version of the address defined like `zrmq3cxYSHSlSQWSPferKnppq_-kTKW6uGf3w5XPUcw` (see https://viewblock.io/arweave/address/zrmq3cxYSHSlSQWSPferKnppq_-kTKW6uGf3w5XPUcw) and not a hex version of it.

Where this hex requirements came from?

Reply:
The hex here is hex(sha256(pubkey)) , I was under the impression that this is how we go from the long version to the short version

Reply:
This has been fixed, the correct format is base64url(sha256(pubkey))